core/primitives/src/crypto.rs

```diff
@@ -603,28 +670,440 @@ pub trait Pair: TypedKey + Sized + Clone + Send + Sync + 'static {
 	fn to_raw_vec(&self) -> Vec<u8>;
 }
 
+/// A type of supported crypto.
+#[derive(Clone, Copy, PartialEq, Eq, Encode, Decode)]
+#[cfg_attr(feature = "std", derive(Debug))]
+#[repr(u32)]
+pub enum Kind {
```

Comment:
The reason to avoid using an enum is that we will never be able to exhaustively define all crypto types. This change makes it impossible to introduce custom crypto without requiring an upstream pull request

Reply:
I'm not convinced it would be otherwise trivial to add new kinds of crypto. Realistically, that crypto would need extern support which would necessitate some alterations to core. However, if it were possible, then it would not be needed any time soon so can go in a separate PR.

Reply:
Why should core need to know anything about any of the kinds of crypto?

Reply:
Mainly for offchain worker stuff. Hopefully we can remove this.